momentumopt/src/momentumopt/setting/PlannerSetting.cpp

```diff
@@ -156,15 +156,50 @@ namespace momentumopt {
         YAML::ReadParameter(planner_vars, "reg_joint_position", reg_joint_position_);
         YAML::ReadParameter(planner_vars, "num_joint_viapoints", num_joint_viapoints_);
         YAML::ReadParameter(planner_vars, "num_base_viapoints", num_base_viapoints_);
+        // Kinematic paramters of second order IK (python)
+        YAML::ReadParameter(planner_vars, "use_second_order_inv_kin", use_second_order_inv_kin_);
```

Comment:
you could use `YAML::ReadParameterDefault(..., false)`. This way the yaml configs don't need to add a new `use_second_order_inv_kin=false` to be compatible with the new code.

Reply:
I understand, but we should not consider the first order IK as default and ideally we should move to second order IK, when it has been tested enough.  Also, anyways we need to read the parameter use_second_order_inv_kin from Yaml file to decide which version of IK to use in python, I intentionally added use_second_order_inv_kin to config file because currently it might be the case that someone prefers to use first order IK or second order IK depending on the demo... So, I think adding use_second_order_inv_kin to the old yaml files and setting it to false makes sense...

Reply:
okay.